.github/workflows/tests.yml

```diff
@@ -402,21 +366,91 @@ jobs:
           ls -lhR benchmarks
           bash ci/scripts/combine-dbs.sh
 
-      - name: Upload benchmark db
-        env:
-          AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
-          AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
-          AWS_DEFAULT_REGION: us-east-2  # this is needed for boto for some reason
-          DB_NAME: benchmark.db
-        run: |
-          aws s3 cp $DB_NAME s3://coiled-runtime-ci/benchmarks/
+      # - name: Upload benchmark db
+      #   env:
+      #     AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
+      #     AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
+      #     AWS_DEFAULT_REGION: us-east-2  # this is needed for boto for some reason
+      #     DB_NAME: benchmark.db
+      #   run: |
+      #     aws s3 cp $DB_NAME s3://coiled-runtime-ci/benchmarks/
 
       - name: Upload benchmark results as artifact
         uses: actions/upload-artifact@v3
         with:
           name: benchmark.db
           path: benchmark.db
 
+  regressions:
+    needs: process-results
+    # Always generate the site, as this can be skipped even if an indirect dependency fails (like a test run)
+    if: always() && github.event_name == 'pull_request' && github.repository == 'coiled/coiled-runtime' #always() && github.ref == 'refs/heads/main' && github.repository == 'coiled/coiled-runtime'
+    name: Detect regressions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: benchmark.db
+
+      - name: Set up environment
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          use-mamba: true
+          python-version: "3.9"
+          environment-file: ci/environment-regressions.yml
+
+      - name: Run detect regressions
+        run: |
+          python detect_regressions.py
+      
+      - name: Create regressions summary
```

Comment:
It seems that this step doesn't run when the previous one failed I might have to use create and artifact and create a separate workflow, not sure yet.

Reply:
You can add an `if: always()` or similar condition to get around that.